docker: multiple users support

tomereli opened the pull request:
Add multiple users support for docker scripts so that multiple users on
the same server can work in parallel running test_flows.sh.

This requires basically renaming all containers to include the username
of the current user.
In addition, we also add the current user prefix to the docker network
which is used for the test, for obvious reasons.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

arnout commented on tools/docker/run.sh:
```diff
@@ -42,7 +42,7 @@ generate_container_random_ip() {
 
 gateway_netid_length() {
-    docker network inspect prplMesh-net --format "{{(index .IPAM.Config 0).Subnet}}" | sed -rn 's/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}(\/[0-9]{2})$/\1/p'
+    docker network inspect prplMesh-net-${SUDO_USER:-${USER}} --format "{{(index .IPAM.Config 0).Subnet}}" | sed -rn 's/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}(\/[0-9]{2})$/\1/p'
```
instead of the username, it makes more sense to pass in a unique ID as an argument (which could default to `${SUDO_USER:-${USER}}`. With that, we can get rid of the docker-in-docker approach and instead run it directly in a shell runner. That would speed up our run-tests CI job significantly because the cache can be used (now it re-downloads the images every time).

If adding the argument is too much work, please at least make it a variable e.g. CONTAINER_NAME_SUFFIX.

tomereli replied:
I should probably already know this - but what do you mean by the docker-in-docker approach?

rmelotte replied:
Because the tests cannot be run in parallel on the same host at the moment, the CI runs the prplmesh-runner containers in a docker container (a new docker daemon is run inside a docker container). It works, but it's a bit overkill and has a few downsides like not keeping the cache and images so we would like to use something else (either a shell runner, or keeping the docker runner but with a single daemon).

tomereli replied:
Understood.

tomereli replied:
Done

arnout reviewed (approved):
I made an important remark but can be fixed easily. And if not, that's fine too, we can fix it later.

rmelotte reviewed (approved):
I agree with Arnout, using an argument/variable would save us time when we get rid of the docker-in-docker approach.

Also if the network/container names are made unique, we really have to ensure that every script that creates a container/network tries to delete it at the end of the test (it's not that important if the username is used).

Commit:
docker: multiple users support

Add multiple users support for docker scripts so that multiple users on
the same server can work in parallel running test_flows.sh.

This requires basically renaming all containers to include the username
of the current user.
In addition, we also add the current user prefix to the docker network
which is used for the test, for obvious reasons.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
gitlab ci: use -u option for test_flows

Previous commit added the --unique option for running tests on the same
server in parallel by multiple users. Use this option in the CI for now.
Later, we will remove the docker-in-docker approach and run tests in
parallel in the server.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
